pkg/api/store/password/store.go

```diff
@@ -189,6 +193,45 @@ func (p *PasswordStore) replacePasswords(sepData, data, existing map[string]inte
 					}
 				}
 			}
+			// delete remaining secrets from existingArr
+			for _, ind := range existingDataMap {
+				if err := p.replacePasswords(convert.ToMapInterface(sepVal), nil, existArr[ind]); err != nil {
```

Comment:
The `existingDataMap` is called at this point left with existing keys that aren't present in the new data, so why do we have to call replacePasswords and not deleteSecrets?